src/workerd/api/sockets.c++

```diff
@@ -9,8 +9,47 @@
 namespace workerd::api {
 
 
+bool isValidAddress(kj::StringPtr address) {
+  // This function performs some basic length and characters checks, it does not guarantee that
+  // the specified address is a valid domain. It should only be used to reject malicious
+  // addresses.
+  if (address.size() > (255 + 6)) {
+    // RFC1035 states that maximum domain name length is 255 octets. But we also add 6 to support
+    // port numbers in the address.
+    //
+    // IP addresses are always shorter, so we take the max domain length instead.
+    return false;
+  }
+
+  for (int i = 0; i < address.size(); i++) {
+    switch (address[i]) {
+      case '-':
+      case '.':
+      case ':':
+      case '_':
+      case '[': case ']': // For IPv6.
+      case '%': // For percent-encoded domain names.
```

Comment:
I don't think hostnames can be percent-encoded.

Reply:
Correct. The whatwg url spec allows for hostname inputs to be percent encoded but will automatically decode. I've encountered a number of cases where folks thought that meant they can percent encoded in general. We should definitely throw an error here and add a test case to catch it